analytical_engine/core/context/java_context_base.h

```diff
@@ -0,0 +1,410 @@
+/** Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ANALYTICAL_ENGINE_CORE_CONTEXT_JAVA_CONTEXT_BASE_H_
+#define ANALYTICAL_ENGINE_CORE_CONTEXT_JAVA_CONTEXT_BASE_H_
+
+#ifdef ENABLE_JAVA_SDK
+
+#include <jni.h>
+#include <iomanip>
+#include <limits>
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "boost/algorithm/string.hpp"
+#include "boost/algorithm/string/split.hpp"
+#include "boost/filesystem/path.hpp"
+#include "boost/property_tree/json_parser.hpp"
+#include "boost/property_tree/ptree.hpp"
+
+#include "grape/app/context_base.h"
+#include "grape/grape.h"
+#include "vineyard/client/client.h"
+#include "vineyard/graph/fragment/fragment_traits.h"
+
+#include "core/config.h"
+#include "core/context/labeled_vertex_property_context.h"
+#include "core/context/vertex_data_context.h"
+#include "core/context/vertex_property_context.h"
+#include "core/error.h"
+#include "core/java/javasdk.h"
+#include "core/object/i_fragment_wrapper.h"
+
+namespace gs {
+static constexpr const char* APP_CONTEXT_GETTER_CLASS =
+    "com/alibaba/graphscope/utils/AppContextGetter";
+static constexpr const char* LOAD_LIBRARY_CLASS =
+    "com/alibaba/graphscope/utils/LoadLibrary";
+static constexpr const char* CONTEXT_UTILS_CLASS =
+    "com/alibaba/graphscope/utils/ContextUtils";
+static constexpr const char* JSON_CLASS_NAME = "com.alibaba.fastjson.JSON";
+static constexpr const char* JVM_OPTIONS_STR =
```

Comment:
I wanted to put these constant in a configure file, but finally I thought to it is ok to write done all const here, since

- There constants are hardly changed.
- `static constexpr` will make sure there are available since compilation, reading a property file introducing possibility for IO error.

Well, If you think read from a configuration file is better, I'm also willing to adapt to your suggestions.

Reply:
If we prefer a configuration file, which format do you think is better for us?